File: 7kyu/sort_numbers.ts
// My solution
export function solution(nums: number[]): number[] {
  if (!nums) {
    return [];
  }

  return nums.sort((a, b) => a - b);
}

/*
Parameters: an array of numbers
Return: an array of numbers or empty
Examples:
  solution([1, 2, 10, 50, 5]) => [1,2,5,10,50]
  solution([]) => []
Pseudocode:
  if !num or nums.length === 0:
    return [];
  return nums.toSorted();
*/

// Top solution
// export const solution = (nums: number[]) => nums.sort((a, b) => a - b);

// Another one (better, IMHO)
// export function solution(nums: number[]): number[] {
//   return nums ? nums.sort((a, b) => a - b) : [];
// }

File: 7kyu/sort_numbers.py
# My solution
def solution(nums):
    if nums is None or nums == []:
        return []
    
    return sorted(nums)

'''
Parameters: a list of numbers
Return: a list
Examples:
    solution([1,2,3,10,5]) -> [1,2,3,5,10]
    solution(None) -> []
    solution([]) -> []
    solution([20,2,10]) -> [2,10,20]
    solution([2,20,10]) -> [2,10,20]
Pseudocode:
    check for none or [] and return []
    return nums.toSorted()
'''

# Top solution
def solution(nums):
    return sorted(nums) if nums else []